### PR TITLE
Provision device with deltas enabled

### LINKED ISSUE #139: Enable Deltas through provisioning

### PR DESCRIPTION
**Pull request by horia-delicoti**
Signed-off-by: Horia Delicoti <horia.delicoti@gmail.com>
Closes #139

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -70,6 +71,9 @@ ava.test.before(async () => {
   console.log(`Downloading device type OS into ${imagePath}`)
   await resinio.downloadDeviceTypeOS(options.deviceType, options.resinOSVersion, imagePath)
 
+  console.log('Enable delta')
```
Comment (jviotti) [inline lib/main.js]: nitty gritty, but should this be `Enabling deltas` to phrase it as the other log comments?

File: lib/main.js [hunk 2]
```diff
@@ -70,6 +71,9 @@ ava.test.before(async () => {
...
+  await resinio.createEnvironmentVariable(options.applicationName, 'RESIN_SUPERVISOR_DELTA', options.delta)
```
Comment (jviotti) [inline lib/main.js]: Maybe we should only create the environment variable if the value was set? ie

```js
if (options.delta) {
    await resinio.createEnvironmentVariable(options.applicationName, 'RESIN_SUPERVISOR_DELTA', options.delta)
}
```

**Comment by jviotti** [comment]
You also need to add this variable to the Makefile, otherwise it will not be bridged.

File: lib/main.js (hunk 1 above)
Comment (horia-delicoti) [reply lib/main.js]: oh yeah, changed.

File: lib/main.js [hunk 3]
```diff
@@ -70,6 +71,11 @@ ava.test.before(async () => {
   console.log(`Downloading device type OS into ${imagePath}`)
   await resinio.downloadDeviceTypeOS(options.deviceType, options.resinOSVersion, imagePath)
 
+  if (options.delta) {
+    console.log('Enabling deltas')
```
Comment (horia-delicoti) [inline lib/main.js]: @jviotti no need to log this if it's not enabled right?

**Comment by horia-delicoti** [comment]
@jviotti No need to add this the env var to Makefile because we have a [for loop](https://github.com/resin-io/resinos-tests/blob/master/Makefile#L15) for all of them

File: lib/main.js (hunk 3 above)
Comment (jviotti) [reply lib/main.js]: Looks good!

**Pull request merged**